scipy/signal/filter_design.py

```diff
@@ -39,6 +39,17 @@ class BadCoefficients(UserWarning):
 abs = absolute
 
 
+def _is_int_type(x):
+    if np.ndim(x) != 0:
+        return False
```

Comment:
Why is this necessary? I think we have been using just `operator.index` as below everywhere else. And `np.array([1]).__index__()` gives a TypeError.

Reply:
It doesn't give a type error in older numpy versions if I remember correctly

Reply:
Should add a comment explaining that though